app/src/main/java/net/geekinpurple/www/appofourown/SearchResults.java

```diff
@@ -54,38 +58,44 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
-    private class Retrieval extends AsyncTask<String, Void, String> {
-        Document doc;
+    @Override
+    public void onListItemClick(ListView l, View v, int position, long id) {
+        Work message = (Work) l.getItemAtPosition(position);
+        Intent i = new Intent(this, WorkInstance.class);
+        i.putExtra("work", message);
+        Log.d("testing", message.toString());
+        startActivity(i);
+    }
 
+    private class Retrieval extends AsyncTask<String, Void, ArrayList<Work>> {
         @Override
-        protected String doInBackground(String... params) {
+        protected ArrayList<Work> doInBackground(String... params) {
             String url = params[0];
             try {
-                Document ao3 =  Jsoup.connect(url).get();
-                Elements works = ao3.getElementsByClass("blurb");
+                Document doc =  Jsoup.connect(url).get();
+                Elements works = doc.getElementsByClass("blurb");
 
                 StringBuffer message = new StringBuffer();
 
-                // Build the array to hand to ListView
-                Vector<Work> worksVector = new Vector<Work>();
+                // Build the array to hand to ArrayAdapter
+                ArrayList<Work> worksVector = new ArrayList<Work>();
                 for (Element work : works) {
                     Work w = new Work(work);
                     worksVector.add(w);
                     message.append(w.toString());
                 }
 
-                String result = message.toString();
-                return result;
+                return worksVector;
             }
             catch (IOException aeiou) {
-                return "Sorry!";
+                return new ArrayList<Work>();
```

Comment:
You should log that an exception happened or something. Just returning an empty list will be hard to diagnose.